packages/suite/src/views/suite/SwitchDevice/DeviceItem/WalletInstance.tsx

```diff
@@ -102,78 +109,82 @@ export const WalletInstance = ({
+                                <MetadataLabeling
+                                    defaultVisibleValue={
+                                        walletLabel === undefined || walletLabel.trim() === ''
+                                            ? defaultWalletLabel
+                                            : walletLabel
+                                    }
+                                    payload={{
+                                        type: 'walletLabel',
+                                        entityKey: instance.state,
+                                        defaultValue: instance.state,
+                                        value: instance?.metadata[
+                                            METADATA_LABELING.ENCRYPTION_VERSION
+                                        ]
+                                            ? walletLabel
+                                            : '',
+                                    }}
+                                    defaultEditableValue={defaultWalletLabel}
+                                />
+                            ) : (
+                                <WalletLabeling device={instance} />
+                            )}
+                        </InstanceType>
+                    )}
+
+                    {!discoveryProcess && (
+                        <InstanceType isSelected={isSelected}>
+                            <Translation id="TR_UNDISCOVERED_WALLET" />
+                        </InstanceType>
+                    )}
+
+                    <InstanceTitle>
+                        <HiddenPlaceholder>
+                            <FiatHeader
+                                size="medium"
+                                fiatAmount={instanceBalance.toString() ?? '0'}
+                                localCurrency={localCurrency}
                             />
-                        ) : (
-                            <WalletLabeling device={instance} />
-                        )}
-                    </InstanceType>
+                        </HiddenPlaceholder>
+                    </InstanceTitle>
+                </Col>
+
+                <Divider />
+
+                {contentType === 'default' && enabled && discoveryProcess && (
+                    <ViewOnly
+                        dataTest={dataTestBase}
+                        setContentType={setContentType}
+                        instance={instance}
+                    />
                 )}
-
-                {!discoveryProcess && (
-                    <InstanceType isSelected={isSelected}>
-                        <Translation id="TR_UNDISCOVERED_WALLET" />
-                    </InstanceType>
+                {contentType === 'ejectConfirmation' && (
+                    <EjectConfirmation
+                        instance={instance}
+                        onClick={e => e.stopPropagation()}
+                        onCancel={e => {
+                            setContentType('default');
```

Comment:
nit multiline, extract to hanld efunction

Reply:
or maybe we can have some halper (higher order function) to add `stopPropagation` to the call


something like:

```tr
const withStopPropagation = (call: ()=> void) => (e: ...) => {
    call();
    e.stopPropagation();
}

// ....

onCancel={withStopPropagation(() => setContentType('default'))}
```